### ISSUE TITLE
Persistent error on HA update 2023.1.5: Retrying setup: 'HassClientResponse' object has no attribute 'status_code'

### ISSUE THREAD

**Issue by hindsight2021** [dylandoamaral/trakt-integration#61]
Error seen in 'integrations' I see others have encountered this recently in other closed issues, deleting/removing from HACS and re-installing did not resolve for me:

**Retrying setup: 'HassClientResponse' object has no attribute 'status_code'**

![image](https://user-images.githubusercontent.com/54675389/212937369-e9a3606b-42a6-482c-8fdc-e4ffcf34aaac.png)

HA Log - error repeats about every 2 mins on startup

2023-01-17 11:11:59.980 ERROR (MainThread) [custom_components.trakt_tv] Unexpected error fetching trakt data: 'HassClientResponse' object has no attribute 'status_code'
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 225, in _async_refresh
self.data = await self._async_update_data()
File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 181, in _async_update_data
return await self.update_method()
File "/config/custom_components/trakt_tv/apis/trakt.py", line 289, in retrieve_data
data = await gather(
File "/config/custom_components/trakt_tv/apis/trakt.py", line 263, in fetch_recommendations
data = await gather(
File "/config/custom_components/trakt_tv/apis/trakt.py", line 256, in fetch_recommendation
return await self.request(
File "/config/custom_components/trakt_tv/apis/trakt.py", line 68, in request
elif response.status_code == 429:
AttributeError: 'HassClientResponse' object has no attribute 'status_code'

**Tried:**
- Removing from HA (deleted)
- Rebooting/reinstalling (no change.)
- Removing again from HA and also HACS
- rebooting & full re-install and Trakt auth, which said successful using the work-around method.
- no change, no sensor is re-added/created

It was working fine prior to today, the only thing I did was update my config YAML to add premieres which I omitted before.

Reverted to your default YAML from install instructions today, no changes.

The only change in my environment was an update to HA: 2023.1.5 done today.

**Comment by dnestico** [comment]
Same

**Comment by manticore1991** [comment]
Yep, same for me. Originally I thought this was due to Trakt's downtime early in December last year, but it appears something else is broken here :(.

**Comment by dylandoamaral** [comment]
I will certainly try to fix this tomorrow night guys !

**Comment by manticore1991** [comment]
Thank you very much, I appreciate it! Thanks also for developing this integration in the first place, it's really awesome! =)

**Comment by hindsight2021** [comment]
@dylandoamaral not sure if it's just me, but I've upgraded to 0.7.3 and it's stopped that specific error, but now it fails to initialize entirely for me. I've tried a full removal from HA/HACS and reinstall, rebooted a few times, same behavior/error in log with debugging enabled:

![image](https://user-images.githubusercontent.com/54675389/214582497-9e1950f9-0c4f-4694-ad29-4b4b784db632.png)

 Logger: homeassistant.config_entries
Source: config_entries.py:825
First occurred: 9:54:46 AM (1 occurrences)
Last logged: 9:54:46 AM
Config entry 'Trakt' for trakt_tv integration not ready yet; Retrying in background

It just seems to be in a startup loop.
It took my Trakt API ID/secret and URL Oauth without issue, it's after this time it bootloops consistently.

any ideas?
